esp-wifi/src/ble/btdm.rs

```diff
@@ -226,9 +218,8 @@ unsafe extern "C" fn queue_recv(queue: *const (), item: *const (), block_time_ms
         block_time_ms
     );
 
-    // is this ticks or millis?
-    let end_time = crate::timer::get_systimer_count()
-        + (block_time_ms as u64 * (crate::timer::TICKS_PER_SECOND / 1000));
+    let end_time_ticks =
+        crate::timer::get_systimer_count() + crate::timer::millis_to_ticks(block_time_ms as u64);
```

Comment:
As ticks, depending on block time and the current systimer value, I think this could overflow a `u64`?

Reply:
not a new bug though and as you noted, timeout handling is otherwise broken